### ISSUE TITLE
the speed problem

### ISSUE THREAD

**Issue by luojing1030** [Dootmaan/DTFD-MIL.PyTorch#3]
Why is it slow to deploy the program to the graphics card, and even reports an error saying that the version is too old, after I have installed the corresponding environment (torch1.4.0, python3.6, scikit-learn0.21.1)? However, other versions of torch load quickly on the graphics card, but an error occurs at the step of "loss1.backward". Have you encountered a similar problem before?

**Comment by luojing1030** [comment]
![1683424107439](https://user-images.githubusercontent.com/87426653/236653555-b02cdadf-db10-43d3-a3d6-e40ce629da3f.jpg)

**Comment by Dootmaan** [comment]
Hi @luojing1030 and thank you for your question. Since our project is mainly based on the official code of DTFD-MIL, running the code also require the use of PyTorch 1.4.0 as the official one does. There are comments in the official code mentioning that change the sequence of loss backward can solve this problem in newer PyTorch versions (which can be found [here]( https://github.com/hrzhang1123/DTFD-MIL/issues/8#issuecomment-1343965040)), but we are not sure whether will that affect the performance.

**Comment by luojing1030** [comment]
Thank you for your help, and also thank you for simplifying the code for the convenience of others.
